Automattic/wp-calypso
# Check against all backup features in isBackupProductIncludedInSitePlan()

#### Changes proposed in this Pull Request

* Complimentary to #61525, this PR addresses a bug where the `<CartPlanOverlapsOwnedProductNotice />` component was not being displayed when a user has a Backup product in their cart when they already have a plan that includes the equivalent backup features.
* This is fixed primarily by adding the new backup features and products (i.e. Backup T1 and T2) to the conditionals used to determine if there is a plan/product overlap.
* Updates `planHasDailyBackup` and `planHasRealTimeBackup` to include an optional flag for checking for superior products, so that these functions can be re-used across the different cart/plan/feature comparison functions. 
* Fixes a translation component that was not displaying correctly.

#### Testing instructions



* Set up a WordPress site with Jetpack, and purchase a plan that includes backup (i.e. Security or Complete).
* Add a backup plan to your cart (i.e. Backup 10 GB).
* Verify the pre-purchase notice is displayed:

<img width="577" alt="Screen Shot 2022-03-02 at 3 53 59 PM" src="https://user-images.githubusercontent.com/10933065/156463899-b9db483d-dc04-4032-8e8e-07c876285e08.png">

* Ensure tests pass: `yarn test-client client/state/sites/products/test/conflicts.ts`



Related to #61577
